includes/Core/Util/Tracking.php

```diff
@@ -117,7 +117,11 @@ public function is_active() {
 	 */
 	private function inline_js_base_data( $data ) {
 		global $hook_suffix;
-		$data['trackingAllowed'] = $this->screens->get_screen( $hook_suffix ) instanceof Screen;
+
+		$screen           = $this->screens->get_screen( $hook_suffix );
+		$tracking_allowed = $screen instanceof Screen;
+
+		$data['trackingAllowed'] = apply_filters( 'googlesitekit_tracking_allowed', $tracking_allowed );
```

Comment:
I understand why this was changed but this wasn't in the IB so let's make sure @felixarntz is on board with it.

Alternatively, we could register proper `Screens` for these even though they're not Site Kit™️ screens but that might be worth considering instead as we have a few other workarounds (such as enqueueing assets) which might be cleaner with a `Screen` as well.

Reply:
Tracking doesn't work on WP dashboard and post list pages without these modifications.

Reply:
Yeah, I forgot that `Screen`s don't really support using for existing/WP-core screens which was my thought there to fit within the existing logic here. I'm not sure we should introduce the ability to disable this for all screens though – that's what the opt-in is for.

Alternatively to using this filter as an on/off switch, we could do something similar to the screen check where we could introduce the concept of "modules with core screens" where those modules return an array of hook-suffixes or `WP_Screen` IDs that we could check for here. That could be useful in other ways as well 🤔  @felixarntz ?

Reply:
I agree with @aaemnnosttv's concern about introducing a filter here. When introducing filters, we should think through whether we really want to open that respective area up for _any_ changes, since it can be done by third parties too.

In this particular case, I think Site Kit should for now remain in control of where our tracking generally works, since we also don't offer APIs do integrate Site Kit with further areas of WordPress.

I'm not sure about what would be the best API to implement here to facilitate this use-case, a generic "module with core screens" approach may also be too broad for only defining where to load tracking. I'd vote for being pragmatic here and add the Idea Hub logic (if the module is active and connected) directly into the `Tracking` class here. Clearly this is not an architecturally clean approach, but we also shouldn't build APIs if we can't assess whether something else in the future will use it or how something else in the future may want to use it - we can always add `TODO` comments to such areas to potentially revisit later.

Reply:
> I'm not sure about what would be the best API to implement here to facilitate this use-case, a generic "module with core screens" approach may also be too broad for only defining where to load tracking.

@felixarntz I think it could have other uses, not only for this. Alternatively, we could leverage the existing `Module_With_Screens` functionality and [the existing logic in place here](https://github.com/google/site-kit-wp/blob/f46f548d709f4224a7af75f40701b3b05c9c848f/includes/Core/Util/Tracking.php#L120) related to it by introducing a new subclass of `Screen` for using with core screens (i.e. screens we are not registering/adding to the menu but we could still hook into `load-$hook_suffix`). Maybe the subclass isn't the best approach here since the suggested new class would actually be simpler than the current `Screen` class. That way we'd only be making the existing API more flexible in a reasonable way for a use-case I think we can see is becoming more relevant as SK starts having a larger footprint in WP admin.

Perhaps this is still out of scope for this issue, so I wouldn't be opposed to doing something inline for the sake of the smallest change that doesn't require added filters/APIs yet and then we can refactor in a follow-up once the new approach is defined.

Reply:
Ok, I have reverted modifications in the Tracking class and leveraged the existing filter to update js inline data. Let me know if it works.